toolkit/tools/graphanalytics/graphanalytics.go

```diff
@@ -0,0 +1,295 @@
+func printIndirectlyClosestToBeingUnblocked(pkgGraph *pkggraph.PkgGraph, maxResults int) {
+	srpmsBlockedBy := make(map[string][]string)
+
+	for _, node := range pkgGraph.AllNodes() {
+		if node.Type != pkggraph.TypeBuild {
+			continue
+		}
+
+		if node.State == pkggraph.StateUnresolved {
+			continue
+		}
+
+		pkgSRPM := filepath.Base(node.SrpmPath)
+
+		search := traverse.BreadthFirst{}
+		search.Walk(pkgGraph, node, func(n graph.Node, d int) (stopSearch bool) {
+			dependency := n.(*pkggraph.PkgNode)
+
+			// Only consider unresolved build nodes.
+			if dependency.State != pkggraph.StateUnresolved && dependency.State != pkggraph.StateBuild {
+				return
+			}
+
+			// Skip requirements that come from the same srpm.
+			dependencySRPM := filepath.Base(dependency.SrpmPath)
+			if pkgSRPM == dependencySRPM {
+				return
+			}
+
+			dependencyName := nodeDependencyName(dependency)
+			insertIfMissing(srpmsBlockedBy, pkgSRPM, dependencyName)
+
+			return
+		})
+
+	}
+
+	printTitle("[INDIRECT] SRPMs closest to being ready to build")
+	printReversedMap(srpmsBlockedBy, "total unmet dependencies", maxResults)
+}
+
+// nodeDependencyName returns a common dependency name for a node that will be shared across similair Meta/Run/Build nodes for the same package.
+func nodeDependencyName(node *pkggraph.PkgNode) (name string) {
+	// Prefer the SRPM name if possible, otherwise use the unversioned package name
+	srpmName := filepath.Base(node.SrpmPath)
+	if srpmName == "" || srpmName == "<NO_SRPM_PATH>" {
+		name = node.VersionedPkg.Name
+	} else {
+		name = srpmName
+	}
+
+	return
+}
+
+// printTitle prints a formatted title
+func printTitle(title string) {
+	logger.Log.Info("")
+	logger.Log.Info("================================================")
+	logger.Log.Info(title)
+	logger.Log.Info("================================================")
+}
+
+// sortMap returns a sorted slice of a map. If inverse is set, it will return the smallest entries first.
+// It will sort entries by the number of values each key has. For keys with the same number of entries, it will
+// sort alphabetically.
+func sortMap(mapToSort map[string][]string, inverse bool) (pairList []mapPair) {
+	pairList = make([]mapPair, 0, len(mapToSort))
+	for key, value := range mapToSort {
+		pairList = append(pairList, mapPair{key, value})
+	}
+
+	sort.Slice(pairList, func(i, j int) bool {
+		if len(pairList[i].value) == len(pairList[j].value) {
+			return pairList[i].key < pairList[j].key
+		}
+
+		if inverse {
+			return len(pairList[i].value) < len(pairList[j].value)
+		}
+
+		return len(pairList[i].value) > len(pairList[j].value)
+	})
+
+	return
+}
+
+// insertIfMissing appens a value to the key in a map if it is not present.
+// Will alter data.
+func insertIfMissing(data map[string][]string, key string, value string) {
+	if sliceutils.Find(data[key], value) == -1 {
+		data[key] = append(data[key], value)
+	}
+}
```

Comment:
Nit (way out of scope, wanted to throw a suggestion maybe for a smaller PR): might be worth to create a separate `StringSet` class with a few methods like `Add`, `Remove`, `Contains`, etc., which could be just a `map[string]bool`. I think we require similar functionality in other places in the code and we just keep re-implementing the same idea with helper functions.